lib/chef/knife/ec2_server_create.rb

```diff
@@ -615,14 +614,22 @@ def bootstrap_for(server, fqdn)
             fqdn = config_value(:fqdn) || fetch_server_fqdn(server.private_ip_address)
           end
         end
-        bootstrap.name_args = [fqdn]
+        name_args = [fqdn]
 
         if config_value(:chef_node_name)
           config[:chef_node_name] = evaluate_node_name(config_value(:chef_node_name))
         else
           config[:chef_node_name] = server.id
         end
-        bootstrap_common_params(bootstrap)
+        bootstrap_common_params
+        register_client
```

Comment:
Looks like a duplication. This should be taken care by parent class.

Reply:
This duplication was fixed in #584.